app/src/main/java/com/duckduckgo/app/browser/ui/HttpAuthenticationDialogFragment.kt

```diff
@@ -0,0 +1,100 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.ui
+
+import androidx.appcompat.app.AlertDialog
+import android.app.Dialog
+import android.os.Bundle
+import android.view.View
+import android.view.WindowManager
+import android.widget.EditText
+import android.widget.TextView
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.browser.model.BasicAuthenticationCredentials
+import com.duckduckgo.app.browser.model.BasicAuthenticationRequest
+import com.duckduckgo.app.global.view.hideKeyboard
+import com.duckduckgo.app.global.view.showKeyboard
+import org.jetbrains.anko.find
+
+class HttpAuthenticationDialogFragment : DialogFragment() {
+
+    var listener: HttpAuthenticationListener? = null
+    lateinit var request: BasicAuthenticationRequest
+
+    interface HttpAuthenticationListener {
+        fun handleAuthentication(request: BasicAuthenticationRequest, credentials: BasicAuthenticationCredentials)
+        fun cancelAuthentication(request: BasicAuthenticationRequest)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val rootView = View.inflate(activity, R.layout.http_authentication, null)
+        val usernameInput = rootView.find<EditText>(R.id.usernameInput)
+        val passwordInput = rootView.find<EditText>(R.id.passwordInput)
+        val informationText = rootView.find<TextView>(R.id.httpAuthInformationText)
+
+        validateBundleArguments()
+
+        val url = arguments!!.getString(KEY_TARGET_URL)
+
+        informationText.text = getString(R.string.authenticationDialogMessage, url)
+
+        val alertBuilder = AlertDialog.Builder(activity!!)
+            .setView(rootView)
+            .setPositiveButton(R.string.authenticationDialogPositiveButton) { _, _ ->
+                listener?.handleAuthentication(request,
+                    BasicAuthenticationCredentials(username = usernameInput.text.toString(), password = passwordInput.text.toString()))
+
+            }.setNegativeButton(R.string.authenticationDialogNegativeButton) { _, _ ->
+                rootView.hideKeyboard()
+                listener?.cancelAuthentication(request)
+            }
+            .setTitle(R.string.authenticationDialogTitle)
+
+        val alert = alertBuilder.create()
+        showKeyboard(usernameInput, alert)
+
+        return alert
+
+    }
+
+    private fun validateBundleArguments() {
+        if (arguments == null) throw IllegalArgumentException("Missing arguments bundle")
+        val args = arguments!!
+        if (!args.containsKey(KEY_TARGET_URL)) {
+            throw IllegalArgumentException("Bundle arguments required [KEY_TARGET_URL]")
+        }
+    }
+
+    private fun showKeyboard(titleInput: EditText, alert: androidx.appcompat.app.AlertDialog) {
+        titleInput.setSelection(titleInput.text.length)
```

Comment:
what's this line needed for?

Reply:
This function was copied from `SaveBookmarkDialogFragment`. I've refactored the `titleInput` parameter to a more generic name and removed this line as the `username` edit text has focus without making this call.

Reply:
Ah, I see. I think it was only added there because those inputs could be pre-populated with the page title and/or URL, and that line would make the cursor jump to the end of the text rather than the start.

But agree, not needed in this case.